### PR TITLE
update: random middle letter for each user

### PR DESCRIPTION
**Pull request by nidhey27**
Update:
- New middle letter for every user
- Show Congratulations pop when user finds 2-3 words in the puzzle

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -165,14 +170,23 @@ export const useMainStore = defineStore({
...
       this.setYesterdaysAnswersAndLastGameDate({ yesterdaysAnswerObj });
 
       // set yesterday and todays answers and letters
-      const { answers, availableLetters, middleLetter } = todaysAnswerObj;
+      const { answers, availableLetters } = todaysAnswerObj;
 
       this.answers = answers;
       this.availableLetters = availableLetters;
-      this.middleLetter = middleLetter;
+      this.middleLetter = selectRandomMiddleLetter(availableLetters); // directly generate random middle letter here
```
Comment (bhavin192) [inline src/store.ts:181]: Nice idea! Though we will have problem with this. Because the logic to select a game from allAnswers.json is still based on the date, all of our users will get same game which has fixed possibilities. 

I suggest this approach instead:
1. We modify the generateAnswerObjs to pick any random array element from the allAnswers. 
2. For yesterdaysAnswerObj we can return null.
3. Here in this store.ts file we can comment out the calls to setYesterdaysAnswersAndLastGameDate

Basically, we remove the notion of date from the game. And for every user the whole game will be different, not just the middle letter.

**Review by bhavin192 — CHANGES_REQUESTED** [review]

File: src/components/GameWon.vue [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+  // Example functionality for the share button
+  if (navigator.share) {
+    navigator.share({
+      title: 'My Spelling Bee Score!',
+      text: `I scored ${store.getUserScore} on Spelling Bee! Can you beat my score?`,
+      url: window.location.href,
+    });
+  } else {
+    alert('Your browser does not support the Share API');
+  }
```
Comment (bhavin192) [inline src/components/GameWon.vue:16]: This is good for now, we can experiment with copying things from https://github.com/infracloudio/react-wordle/blob/main/src/lib/share.ts later.

Comment (nidhey27) [reply src/components/GameWon.vue:16]: Yup!

File: src/store.ts (hunk 1 above)
Comment (nidhey27) [reply src/store.ts:181]: Reverted!
